src/main/java/com/dream11/aerospike/client/AerospikeClientImpl.java

```diff
@@ -0,0 +1,343 @@
+package com.dream11.aerospike.client;
+
+import com.aerospike.client.*;
+import com.aerospike.client.async.EventLoops;
+import com.aerospike.client.cluster.ClusterStats;
+import com.aerospike.client.policy.*;
+import com.aerospike.client.query.KeyRecord;
+import com.aerospike.client.query.PartitionFilter;
+import com.aerospike.client.query.Statement;
+import com.dream11.aerospike.listeners.BatchListListenerImpl;
+import com.dream11.aerospike.listeners.DeleteListenerImpl;
+import com.dream11.aerospike.listeners.ExecuteListenerImpl;
+import com.dream11.aerospike.listeners.ExistsArrayListenerImpl;
+import com.dream11.aerospike.listeners.ExistsListenerImpl;
+import com.dream11.aerospike.listeners.QueryResultListenerImpl;
+import com.dream11.aerospike.listeners.RecordArrayListenerImpl;
+import com.dream11.aerospike.listeners.RecordListenerImpl;
+import com.dream11.aerospike.listeners.WriteListenerImpl;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.impl.VertxInternal;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class AerospikeClientImpl extends com.aerospike.client.AerospikeClient implements AerospikeClient {
+  final VertxInternal vertx;
+  private EventLoops eventLoops;
+
+  public com.aerospike.client.AerospikeClient getAerospikeClient() {
+    return this;
+  }
+
+  public AerospikeClientImpl(Vertx vertx, ClientPolicy policy, Host... hosts) {
+    super(policy, hosts);
+    this.vertx = (VertxInternal) vertx;
+    eventLoops = policy.eventLoops;
+  }
+
+  private  <T> void schedule(Handler<Promise<T>> handler, Handler<AsyncResult<T>> resultHandler) {
+    vertx.getOrCreateContext().executeBlocking(promise -> {
+      try {
+        handler.handle((Promise<T>)promise);
+      } catch (AerospikeException e) {
+        promise.fail(e);
+      }
+    }, result -> resultHandler.handle((AsyncResult<T>) result));
+  }
+
+  /**
+   * Determine if we are ready to talk to the database server cluster.
```

Comment:
I think we should keep the documentation about the method and its parameters in the interface itself. In the implementation, we should only put details of implementation.